### ISSUE TITLE
A11y_.NET Core_ASP.NETCore_BlazorWasmManageyouraccountTwoPersonalData_DevTools:Incorrect role type provided to the "Delete" button as link.

### ISSUE THREAD

**Issue by v-akr** [dotnet/aspnetcore#42362]
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

### GitHubTags:
#A11yMAS;#A11yWCAG2.1;#A11yTCS;#.NET Core[aspnetcore]-Win32-June2022;#.NET Core;#A11ySev2;#WCAG4.1.2;#Win11;#Benchmark;#Zoom

### Environment Details:
.NET Core
ASP.NET Core
Operating System: Windows 11 Enterprise 21H2

### Prerequisite:
Set resolution to 1280*1024

### Steps to Reproduce:

1. Open command prompt
2. Create a folder using command "mkdir foldername" e.g. "mkdir BlazorWasmr"
3. Move to the created directory using command "cd foldername"
4. Run the following command "dotnet new blazorwasm --hosted -au Individual" and hit enter.
5. Run command "cd Server" and hit Enter.
6. Run the following command "dotnet run"
7. Copy the First URL link from Command prompt and copy it
8. Paste it in URL section of Browser and hit Search
9. Browser page will open with above template loaded
10. Register account then login.
11. Tab till "Hello with the userid" and click on it
12. Navigate to Personal data and click on it.
13. Navigate to the delete button and inspect the control type.

### Actual:
Incorrect role type provided to the "Delete" button as link.

**Similar issue is observed with below scenario :** 

**1)ASP.NET Core - ManageyouraccountpersonalData :** 

1.  Open command prompt
2. Create a folder using command "mkdir foldername" e.g. "mkdir angular"
3.  Move to the created directory using command "cd foldername"
4. Run the following command "dotnet new angular -au Individual" and hit enter.
5. Run the following command "dotnet run"
6. Copy the First URL link from Command prompt and copy it
7. Paste it in URL section of Browser and hit Search
8. Browser page will open with above template loaded
9. To get access to all the pages you'll need to register first and then log-in.
10. Login with your registered account
11. Tab till "Hello with the userid" and click on it.
12. Navigate to Personal data and hit enter.
13. Tab till "Delete" button and inspect

### Expected:
Correct role type should be provided to the "Delete" button as Button.

### User Impact:
Screen reader users will get incorrect information about the control.

![image](https://user-images.githubusercontent.com/106580806/175870280-e5c8bf83-8b94-44b1-bb34-5b7764748ac9.png)

**Comment by HaoK** [comment]
I'm guessing this is because the button gets rendered like so:

```
<button class="w-100 btn btn-lg btn-danger" type="submit">Delete data and close my account</button>

<a id="delete" class="btn btn-danger" href="[/Identity/Account/Manage/DeletePersonalData](https://localhost:44423/Identity/Account/Manage/DeletePersonalData)">Delete</a>
```

Unlike the primary button which is a button:
```
<button class="btn btn-primary" type="submit">Download</button>
```

**Comment by HaoK** [comment]
Actually this is correct, its a link not a button... since it redirects to the actual delete Page:

Its just styled like a button:
```
<a id="delete" asp-page="DeletePersonalData" class="btn btn-danger">Delete</a>
```

![image](https://user-images.githubusercontent.com/6537861/175656383-b2535440-40fa-4662-95da-e45bdc017b59.png)

Clicking on delete doesn't delete, it goes to

![image](https://user-images.githubusercontent.com/6537861/175656424-035d5d6f-e4c6-4fd9-9491-496077f137fc.png)

This behavior seems fine @blowdart thoughts?

**Comment by adityamandaleeka** [comment]
@merriemcgaw Please let us know how we should proceed with this one to meet our accessibility standards.

**Comment by merriemcgaw** [comment]
@v-akr , you have tagged this as a Win32 app, but it is really a web app. Is there any change to the evaluation knowing it uses web rules?

**Comment by merriemcgaw** [comment]
@HaoK @adityamandaleeka , 

I did some reading of the requirements and because you're taking a specific action with the button, it should be labeled as such. From a screen reader user's perspective, the links on the side are navigational, so they should be links. The links in the page Download and Delete are taking action and therefore the screen readers should think of them as a button.

**Comment by HaoK** [comment]
Okay, what does that actually mean, we just need to add additional role metadata to this button?

**Comment by merriemcgaw** [comment]
Exactly. I believe you can set the aria-role to button even with the element being a link.

**Comment by mkArtakMSFT** [comment]
@merriemcgaw what are the expectations for validation on these? This seems to have been open for almost a month now.

**Comment by merriemcgaw** [comment]
I will reach out to the testing team - it should have been done by now. They'll pick it up shortly. Thanks for the reminder. I'll look for others in the same state so we can get all of them cleaned up at the same time.

**Comment by Swati1700** [comment]
@merriemcgaw We downloaded the latest RC1 build from the first column of the above table and verified the bug, and the bug still reproduces.
Please find the attachment of the version
![image](https://user-images.githubusercontent.com/86710888/185593964-6d70270c-eb5a-4773-a260-2eb9a63c7da1.png)
Could you please provide the repro steps that you followed, if the issue is not reproducing at your end.

**Comment by merriemcgaw** [comment]
@HaoK is this another case of just needing to wait for the code to flow to installers? This one is definitely older though, which is why I ask.

**Comment by HaoK** [comment]
I see the following using todays build:

It has the `<a id="delete" role="button" class="btn btn-danger" href="[/Identity/Account/Manage/DeletePersonalData](https://localhost:44456/Identity/Account/Manage/DeletePersonalData)">Delete</a>`

Maybe they aren't uninstalling all the old net 7 sdks?  

7.0.100-rc.2.22417.1 [C:\Program Files\dotnet\sdk] is the version I am using from the dotnet/installers main page

![image](https://user-images.githubusercontent.com/6537861/185711034-f831135c-2226-45c1-8891-ea4862de8708.png)

**Comment by merriemcgaw** [comment]
I have questions about whether this needs to have a button role at all. @rperez030 if something that visually looks like a button, but is really a link moving to a new page, would it be OK to have it reported as a link? In this case it redirects to the "delete personal info" page where the actual deletion takes place.

**Comment by Swati1700** [comment]
We have installed latest build dotnet-sdk-7.0.100-rc.2.22452.3-win-x64 but getting a below error.
![image](https://user-images.githubusercontent.com/86710888/188159777-7ae97691-cab5-4069-8d3b-2dc0be67146d.png)

**Comment by merriemcgaw** [comment]
@HaoK is this the blocking bug we thought was fixed? Let's track where that might be, I really hope it makes it into the RC2 SDK

**Comment by HaoK** [comment]
This just looks like some kind of nuget restore issue that isn't really related, I don't know if the packages aren't on the feed or if they are missing some nuget sources

**Comment by HaoK** [comment]
They probably need this feed:
```
<configuration>
  <packageSources>
    <add key="dotnet7" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7/nuget/v3/index.json" />
  </packageSources>
</configuration>
```

**Comment by merriemcgaw** [comment]
@HaoK would you be so kind as to confirm that will resolve the situation and provide exact repro steps for the testers?

**Comment by HaoK** [comment]
Its hard for me to provide exact repro steps as I only have a dev machine, so all I do is install the sdk open a powershell and dotnet new angular.  But I'm not sure what ambient settings/configuration I have setup due to day to day work/etc.

If they want to give me remote access to a machine to install the builds and verify things on their machine before having them try, I'm happy to do that, but I don't really do anything other than install the sdk and then follow the steps in the bug

**Comment by merriemcgaw** [comment]
I can try to build a clean VM and give you both access to it if it works properly.

**Comment by merriemcgaw** [comment]
@v-akr - the updated machine setup instructions are as follows:
1. On a clean machine install .NET RC2 SDK from the **Top, Middle** cell in the [Installer Table](https://github.com/dotnet/installer#table)
2. Open a command prompt and enter the following command: dotnet nuget add source -n dotnet7 https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7/nuget/v3/index.json
3. Follow the test repro steps from here.

**Comment by mkArtakMSFT** [comment]
Moving this to the rc2 milestone to make sure the team watches this.

**Comment by merriemcgaw** [comment]
@Swati1700 @v-akr please retest all of the ASP.NET Core scenarios with the latest instructions.

**Comment by v-akr** [comment]
![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/106580806/192531037-e3538193-3297-4e4c-bc9d-b70da646600a.png)

Verified the issue and the issue is fixed. Hence closing the bug.

**Comment by mkArtakMSFT** [comment]
Issue is fixed. Adding #Closed; tag.
